### ISSUE TITLE
Issue 012: Implement unlabeled attribute-volume dataset for MAE pretraining

### ISSUE THREAD

**Issue by ozw4** [ozw4/seis-attr-ssl#17]
# Issue 012: Implement unlabeled attribute-volume dataset for MAE pretraining

## Context

The MAE pretraining model will consume randomly sampled local/context crops from NOPIMS manifests. This issue creates the dataset object that returns MVP-compatible sample dictionaries. It should rely on the registry, manifest schema, memmap volume store, and crop sampler from earlier issues.

F3 must not be used in this dataset.

## Scope

Implement the unlabeled pretraining dataset.

Recommended files:

```text
src/seis_attr_ssl/data/pretrain_dataset.py
src/seis_attr_ssl/data/attribute_subset.py
tests/test_pretrain_dataset.py
```

## Dataset output contract

For each sample, return a dictionary with:

```python
{
    "x": np.ndarray,                    # [C, 128, 128, 128], float32
    "target": np.ndarray,               # [A, 128, 128, 128], float32, A=10
    "attribute_ids": np.ndarray,         # [C], int64
    "target_attribute_ids": np.ndarray,  # [A], int64, [0..9]
    "valid_attributes": np.ndarray,      # [C], bool
    "target_valid": np.ndarray,          # [A], bool
    "coords": dict[str, object],
    "context": np.ndarray | None,        # [C, 128, 128, 128] if enabled
    "context_valid_mask": np.ndarray | None,
    "local_valid_mask": np.ndarray,      # [128, 128, 128], bool
}
```

Use NumPy arrays for now. PyTorch conversion can be added in the training phase.

## Attribute subset behavior

Implement a simple subset sampler:

```python
sample_attribute_subset(
    available_attribute_ids: Sequence[int],
    min_input_attributes: int,
    max_input_attributes: int,
    rng: np.random.Generator,
) -> tuple[int, ...]
```

Requirements:

- Always include `amplitude_norm` if available.
- Sample between `min_input_attributes` and `max_input_attributes`.
- Respect the fixed registry IDs.
- If not enough attributes are available, either raise a clear error or return all available attributes according to a documented config flag.

## Dataset construction

Implement something like:

```python
class NopimsAttributePretrainDataset:
    def __init__(
        self,
        manifests: Sequence[SurveyManifest],
        registry: AttributeRegistry,
        local_crop_size_xyz=(128, 128, 128),
        context_crop_size_xyz=(512, 512, 512),
        context_downsample=4,
        use_context=True,
        min_input_attributes=4,
        max_input_attributes=10,
        seed=42,
        samples_per_epoch=None,
    ):
        ...
```

`__len__` should use `samples_per_epoch` if provided; otherwise use a deterministic manifest-based default.

## Acceptance criteria

- Dataset samples have exact MVP shapes.
- `target` is always ordered by the full MVP registry.
- `x` uses a sampled subset and corresponding `attribute_ids`.
- Context, if enabled, is downsampled to `[128,128,128]`.
- Dataset is deterministic for a fixed seed and index.
- F3 paths or F3 configs are not used.

## Tests

Use temporary synthetic manifests with `.npy` arrays. Verify:

- sample keys
- sample shapes
- attribute ID ordering
- target ordering
- context shape
- deterministic sampling
- missing required amplitude behavior
- available subset handling

Run:

```bash
python -m compileall -q src proc tests
PYTHONPATH=src pytest -q tests/test_pretrain_dataset.py
```

## Non-goals

- Do not implement spatial/attribute MAE masking here; that is the next masking phase.
- Do not implement PyTorch `DataLoader` collation here unless needed for a minimal smoke test.
- Do not implement F3 fine-tuning dataset here.
